tests/test_scenarios.py

```diff
@@ -0,0 +1,173 @@
+
+
+# launch a server with subprocess from local_server.py
+@pytest.mark.usefixtures("httpd_server_base", "store_info_sets")
+def test_scenarios(
+    httpd_server_base: str,
+    store_info_sets,
+):
+    assert httpd_server_base
+
+    for config in Path(CONFIGS).glob("*.yml"):
+        if config.name not in SCENARIOS_OUTCOMES:
+            log.debug(f"Skipping scenario {config}")
+            continue
+        for test_case in SCENARIOS_OUTCOMES[config.name]:
+            mode = test_case["mode"]
+            log.debug(f"{mode=}")
+
+            # TODO: when mode is added in 0.0.5 add tests here
+            # for now continue
+
+            if mode == "nostop":
+                log.debug(f"Skipping scenario {config} with mode {mode}")
+                continue
+
+            for store_info in store_info_sets:
+                log.debug(f"Running scenario {config} with store {store_info}")
+                travharv = TravHarv(
+                    config,
+                    store_info,
+                )
+                travharv.process()
+                # assert travharv.error_occurred
+                # other assertions here based on the scenarios
+
+                # make connection with the store
+                # create RDF_store
+                core_store: RDFStore = create_rdf_store(*store_info)
+                rdf_store_access = RDFStoreAccess(core_store)
+
+                # get the context name for the config name
+                context_name_graph = rdf_store_access._nmapper.key_to_ng(
+                    config.name
+                )
+                log.debug(f"{context_name_graph=}")
+
+                # make sparql for the test
+                sparql = QUERY_BUILDER.build_syntax(
+                    "execution_report_data.sparql",
+                    context_name_graph=context_name_graph,
+                )
+
+                log.debug(f"{store_info=}")
+
+                # if store info is an empty tuple then skip
+                if store_info == ():
+                    sparql = QUERY_BUILDER.build_syntax(
+                        "execution_report_data.sparql",
+                    )
+                    execution_report_data_result = (
+                        travharv.target_store.select(sparql)
+                    )
+                    sparql_all = "SELECT ?s ?p ?o WHERE { ?s ?p ?o }"
+                    all_triples = travharv.target_store.select(sparql_all)
+                else:
+                    execution_report_data_result = rdf_store_access.select(
+                        sparql
+                    )
+                    sparql_all = (
+                        "SELECT ?s ?p ?o FROM <"
+                        + context_name_graph
+                        + "> \n WHERE { ?s ?p ?o }"
+                    )
+                    all_triples = rdf_store_access.select(sparql_all)
+                # get length of the result
+                log.debug(f"{len(execution_report_data_result)=}")
```

Comment:
i think getting this length is the whole purpose of this way of working....

but it feels more complex then needed

vague memory of the triples of that report being generated through some pysubyt kind of working? if so, can't that graph be reproduced and it length be calculated?

also the object that produces that can probably be smartly be "counted recursively" to detect that same count of expected triples?

and not only that -- given this is a test, it should be predictable what that length needs to be ?

Reply:
These were indeed being generated , but only the last retriebed graph was added to the execution report since that wa the only one of use for laurian. We can revisited this and add all the properties of the graphs retrieved so I can use the count recursivly property of the execution report?

This does not solve the issue that I have about all the comment I wrote abou the tests and we will still need to talk about this with @laurianvm